### ISSUE TITLE
Make Comment Form sensitive

### ISSUE THREAD

**Issue by nerdykit** [ZenPhoto20/ZenPhoto20#801]
albums.php and image.php both display "Comments closed" and "no comments" when comments are off and there are no comments to display.  My suggestion is that if there are no comments in the database for this page and comments are turned off then there should not be any messages appearing on the screen.  ie: make the commentformdisplay function a bit smarter in what it does or does not display.  Not a bug, just a suggestion.  For now I've just edited the theme pages to remove the call to the function but it would be more elegant to have a "smart" function.

Thanks

NK

**Comment by sbillard** [comment]
I totally agree, in fact I believe this is how things actually work. Anyway, that is the way the code reads and what happens on my test site. If there are no comments recorder and comments are not allowed, the printCommentForm() function does nothing.

I do note that three themes may display "no comments" on some pages. These are zenFlkuid, zoBoiotstrap, and zoArdoise. But their test seems correct and I do not see the text on pages with no comments disabled and no comments in the DB.

"Closed for comments" appears only in the printCommentForm() function and is gated with the test for comments enabled and comments in the database. "Comments closed" appears nowhere in my searches.

So, where are you seeing these messages inappropriately? (Also, are you running the 1.7 release?)

**Comment by nerdykit** [comment]
I'm using 17.0.0.01     Actually I just went and removed my 'comment out' slashes from album.php and image.php and looked again and there's no messages appearing about comments closed or anything else relating to comments when those pages are viewed.  If I ever see one of these lines appearing on the site again I'll do a screen shot.  I tried it both as an admin user and a regular viewer.  Both with the same non-result.  Will close this request as completely unnecessary.  Thanks.
